src/carousel/index.ts

```diff
@@ -0,0 +1,221 @@
+import { alias, tag, utils, Tag } from '@storefront/core';
+
+const MIN_SWIPE_DISTANCE = 20;
+
+@alias('carousel')
+@tag('gb-carousel', require('./index.html'), require('./index.css'))
+class Carousel {
+  refs: {
+    wrapper: HTMLDivElement,
+    track: HTMLDivElement
+  };
+  props: Carousel.Props = { items: [] };
+  state: Carousel.State = {
+    settings: {
+      speed: 0,
+      slidesToShow: 1,
+      slidesToScroll: 1
+    },
+    currentSlide: 0,
+    transitioning: false,
+    items: []
+  };
+  animationEndCallback: NodeJS.Timer | null = null;
+
+  init() {
+    this.state.settings = { ...this.state.settings, ...this.props.settings };
+  }
+
+  onMount() {
+    utils.WINDOW().addEventListener('resize', this.update);
+  }
+
+  onUpdate() {
+    this.state.items = this.cloneItems();
+  }
+
+  onUnMount() {
```

Comment:
Should be onUnmount instead, onUnMount will never be called otherwise

You can see which are valid at storefront-core/src/core/tag

Reply:
good catch